cdblib/cdbdump.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import print_function
+
+import argparse
+import sys
+
+import six
+
+import cdblib
+
+
+def cdbdump(parsed_args, **kwargs):
+    # Read binary data from stdin by default
+    stdin = kwargs.get('stdin')
+    if stdin is None:
+        stdin = sys.stdin if six.PY2 else sys.stdin.buffer
+
+    # Print text data to stdout by default
+    stdout = kwargs.get('stdout')
+    if stdout is None:
+        stdout = sys.stdout if six.PY2 else sys.stdout.buffer
+
+    # Consume stdin and parse the cdb file
+    reader_cls = cdblib.Reader64 if parsed_args['64'] else cdblib.Reader
+    data = stdin.read()
+    reader = reader_cls(data)
+
+    # Dump the file's contents to the ouput stream
+    for key, value in reader.iteritems():
+        item = (
+            b'+',
+            str(len(key)).encode('ascii'),
+            b',',
+            str(len(value)).encode('ascii'),
+            b':',
+            key,
+            b'->',
+            value,
+            b'\n',
+        )
+        stdout.write(b''.join(item))
+
+    # Print final newline
+    stdout.write(b'\n')
+
+
+def main(args=None, **kwargs):
+    args = sys.argv[1:] if (args is None) else args
+    parser = argparse.ArgumentParser(
+        description=(
+            "Python version of djb's cdbdump. "
+            "Supports standard 32-bit cdb files as well as 64-bit variants."
+        )
+    )
+    parser.add_argument(
+        '-64', action='store_true', help='Use non-standard 64-bit file offsets'
```

Comment:
I went with `-64` as the command line flag for large file support here; I figured that's a bit more intuitive than `-8`. I'll probably change the other script to use the same argument.

Reply:
Does a 2 character short option work? I've never seen one before

Reply:
It does. You can't put it in line with other short ones (e.g. `ls -la` instead of `ls -l -a`).

I could be convinced to change back to `-8`, since this is a little exotic.

```python
>>> import argparse
... 
... parser = argparse.ArgumentParser()
... parser.add_argument('-64', action='store_true')
... parser.parse_args(['-64'])
Namespace(**{'64': True})
```

```bash
$ <./tests/testdata/top250pws.cdb64 python-pure-cdbdump -64
+6,1:123456->1
+8,1:password->2
+5,1:phpbb->3
+6,1:qwerty->4
...
```

Reply:
I kinda like it :) Ship it!